vowpalwabbit/core/tests/simulator.cc

```diff
@@ -94,6 +95,7 @@ std::pair<std::string, float> cb_sim::get_action(VW::workspace* vw, const std::m
   auto const& scores = examples[0]->pred.a_s;
   std::vector<float> ordered_scores(scores.size());
   for (auto const& action_score : scores) { ordered_scores[action_score.action] = action_score.score; }
+  pmf.reserve(ordered_scores.size());
```

Comment:
Interesting optimization.  Why not declare the vector here and use the copy constructor?

Reply:
Yeah could do that too, would be better. This was just me applying a clang-tidy suggestion as I go.